runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -844,14 +839,23 @@ public void txAbort() {
     public void close() {
         if (TransactionalContext.isInTransaction()) {
             AbstractTransactionalContext rootContext = TransactionalContext.getRootContext();
-            rootContext.setTxnContext(null);
```

Comment:
Just curious why this needs to be removed.

Reply:
just wanted to put all the clear and resets of the root context's data structures in one place

Reply:
Do we also need to remove L834 in that case inside txAbort()?

Reply:
If a transaction is committed successfully, where would it set `rootContext.setTxnContext(null);`? or doesn't it matter to not set it to null? I'm not familiar with this part.

Reply:
I think the Autoclosable pattern takes care of calling the close() method. Apart from that, we also call the removeTransaction() from ObjectView.TXEnd()

Reply:
Autoclosable calls close() only when it's inside try-with-resource which makes it insufficient in all use cases. But as you mentioned, removeContext() should be enough in this case.